### ISSUE TITLE
Rule proposal: newline between props and methods definitions

### ISSUE THREAD

**Issue by domcorvasce** [vuejs/eslint-plugin-vue#1657]
**Please describe what the rule should do:**

Enforce newline between props and methods definitions.

**What category should the rule belong to?**

[x] Enforces code style (layout)
[ ] Warns about a potential error (problem)
[ ] Suggests an alternate way of doing something (suggestion)
[ ] Other (please specify:)

**Provide 2-3 code examples that this rule should warn about:**

```js
// BAD
props: {
  // Latitude
  lat: {
    type: Float,
    required: true,
  },
  // Longitude
  lon: {
    type: Float,
    required: true,
  },
},

// GOOD
props: {
  // Latitude
  lat: {
    type: Float,
    required: true,
  },
  
  // Longitude
  lon: {
    type: Float,
    required: true,
  },
},
```

```js
// BAD
computed: {
  lat() {
    this.getProcessedLatitude();
  },
  lon() {
    this.getProcessedLongitude();
  },
},

// GOOD
computed: {
  lat() {
    this.getProcessedLatitude();
  },
  
  lon() {
    this.getProcessedLongitude();
  },
},
```

```js
// BAD
methods: {
  /**
    * Returns processed latitude coordinates.
    *
    * @returns {Float}
    */
  getProcessedLatitude() {
    // ...
  },
  /**
    * Returns processed longitude coordinates.
    *
    * @returns {Float}
    */
  getProcessedLongitude() {
    // ...
  },
},

// GOOD
methods: {
  /**
    * Returns processed latitude coordinates.
    *
    * @returns {Float}
    */
  getProcessedLatitude() {
    // ...
  },
  
  /**
    * Returns processed longitude coordinates.
    *
    * @returns {Float}
    */
  getProcessedLongitude() {
    // ...
  },
},
```

**Additional context**
I think this could improve code readability, especially with long methods.
[Black](https://black.readthedocs.io/) already includes similar checks for Python code.

**Comment by inker** [comment]
Have you tried [vue/new-line-between-multi-line-property](https://eslint.vuejs.org/rules/new-line-between-multi-line-property.html)?

**Comment by DENCREAT** [comment]
@inker that's not the same.
if you use your provided rule multiline properties in data object or any another multiline objects will also be split with empty lines.
What if I'd like to add empty lines only between Vue component properties for the first level properties like computed, methods or just between props but I don't need to have empty lines between multiline properties in a data object.

**Comment by FloEdelmann** [comment]
There is also https://eslint.vuejs.org/rules/padding-lines-in-component-definition.html. Both rules together should do exactly what you intend.
